vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -183,7 +181,23 @@ type internal FSharpProjectOptionsManager
     /// Get the options for a project
     member this.TryGetOptionsForProject(projectId: ProjectId) = 
         match projectTable.TryGetValue(projectId) with
-        | true, ((_referencedProjects, options), _) -> Some options 
+        | true, ((_referencedProjects, options), _) ->
+            let (sourcePaths,_,otherOptions) = this.CommandLineOptions.GetOptionsWithProjectId(projectId)
+            if sourcePaths.Length = 0 || otherOptions.Length = 0 then
+                Some options
+            else
+                Some { ProjectFileName                   = options.ProjectFileName
+                       SourceFiles                       = if sourcePaths.Length = 0 then options.SourceFiles else sourcePaths
```

Comment:
You already know from the preceding condition that ``sourcePaths.Lengths = 0`` is false, you can simplify this

Reply:
The other line is an 'or' || so I don't necessarily know that it is 0.
````
            if sourcePaths.Length = 0 || otherOptions.Length = 0 then
````

Reply:
ok